src/main/java/hudson/plugins/git/util/BuildData.java

```diff
@@ -57,7 +57,7 @@ public BuildData(String scmName) {
         this.scmName = scmName;
     }
 
-    public BuildData(String scmName, Collection<UserRemoteConfig> remoteConfigs) {
+    public BuildData(String scmName, List<UserRemoteConfig> remoteConfigs) {
```

Comment:
I don't think that is a safe change for API compatibility, is it?  You've changed the signature of the constructor.  I believe that callers who previously passed a Set to the BuildData constructor will now be broken.

Can you explain why you changed from a Collection to a List?